include/grpc/grpc_security_constants.h

```diff
@@ -105,6 +106,28 @@ typedef enum {
   GRPC_SSL_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY
 } grpc_ssl_client_certificate_request_type;
 
+
+typedef enum {
+  /** Default option: performs server certificate verification and hostname
+     verification */
+  GRPC_SSL_SERVER_VERIFICATION,
+  /** Performs server certificate verification, but skips hostname verification
+   */
+  GRPC_SSL_SKIP_HOSTNAME_VERIFICATION,
+  /** Performs hostname name verification, but skips server certificate
+     verification */
+  GRPC_SSL_SKIP_SERVER_CERTIFICATE_VERIFICATION,
+  /** Skips both server certificate and hostname verification */
+  GRPC_SSL_SKIP_ALL_SERVER_VERIFICATION
+} grpc_ssl_server_verification_option;
+
+typedef enum {
+  /** Get peer leaf certificate */
+  GRPC_SSL_PEER_LEAF_CERTIFICATE,
+  /** Get peer full chain */
+  GRPC_SSL_PEER_FULL_CHAIN
+} grpc_ssl_peer_cert_request_type;
```

Comment:
Is it used anywhere? Another option is to provide two TSI properties: one for full cert chain and the other for leaf cert and pass both of them to server authorization check where you can do your own custom check.

Reply:
Actually, it's not needed in experimental namespace as we will always return the full chain as reflected in peer_pem documentation of server authorization check. 

It's an artifact from earlier PR where we were trying to have a backwards compatibility. 

I will remove it.